### PR TITLE
Transcend Koa to multidimensional beings, also misc improvements

### PR DESCRIPTION
**Pull request by Corosauce**

### CODE REVIEW

File: src/main/java/net/tropicraft/core/common/entity/passive/EntityKoaBase.java [hunk 1]
```diff
@@ -672,6 +707,78 @@ public void setFightingItem() {
...
+
+        if (getVillage() == null) {
+            TownKoaVillage village = getVillageWithinRange(128);
+            if (village != null) {
+                dbg("Koa found a village to join: " + village.locationID);
+                this.setVillageAndDimID(village.locationID, village.dimID);
+            } else {
+                //check we have a chest locked in
+                TileEntity tile = world.getTileEntity(getHomePosition());
+                if ((tile instanceof TileEntityChest)) {
```
Comment (Cojomax99) [inline src/main/java/net/tropicraft/core/common/entity/passive/EntityKoaBase.java:737]: probably need a tile != null check here

File: src/main/java/net/tropicraft/core/common/event/MiscEvents.java [hunk 2]
```diff
@@ -79,12 +79,13 @@ public void tickServer(ServerTickEvent event) {
 
     @SubscribeEvent
     public void tickServer(TickEvent.WorldTickEvent event) {
-        if (event.world.provider instanceof WorldProviderTropicraft) {
+        //tick for all dims now!
```
Comment (Cojomax99) [inline src/main/java/net/tropicraft/core/common/event/MiscEvents.java:82]: ALL THE DIMS!

File: src/main/java/net/tropicraft/core/common/entity/passive/EntityKoaBase.java (hunk 1 above)
Comment (Corosauce) [reply src/main/java/net/tropicraft/core/common/entity/passive/EntityKoaBase.java:737]: should be fine without, wont NPE for instance check, and i only use if its TileEntityChest

**Review by Cojomax99 — APPROVED** [review]
:chestnut: :tada:

**Pull request merged**